### PR TITLE
Adjust e2e tests to verify deletion of resources

### PR DESCRIPTION
**Pull request by sutaakar**
# Issue link

# What changes have been made

Expand e2e tests to verify that resource deletion is successful.
Side reason for this change is to make sure e2e tests are stable as it seems that deleting Workload resources, local queue and namespace at the same time can lead to race condition when ClusterQueue is not properly updated, causing Workloads to hang.

# Verification steps

Verified by running e2e tests locally.

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: test/e2e/mnist_rayjob_raycluster_test.go [hunk 1]
```diff
@@ -81,20 +81,22 @@ func runMnistRayJobRayCluster(t *testing.T, accelerator string, numberOfGpus int
...
 		To(WithTransform(RayJobStatus, Equal(rayv1.JobStatusSucceeded)))
+
+	WriteRayJobAPILogs(test, rayClient, GetRayJobId(test, rayJob.Namespace, rayJob.Name))
+
+	test.T().Logf("Deleting RayCluster %s/%s", rayCluster.Namespace, rayCluster.Name)
+	err = test.Client().Ray().RayV1().RayClusters(namespace.Name).Delete(test.Ctx(), rayCluster.Name, metav1.DeleteOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+
+	test.T().Logf("Waiting for RayCluster %s/%s to be deleted", rayCluster.Namespace, rayCluster.Name)
+	test.Eventually(RayClusters(test, namespace.Name), TestTimeoutLong).Should(BeEmpty())
```
Comment (jiripetrlik) [inline test/e2e/mnist_rayjob_raycluster_test.go:99]: Not sure if "TestTimeoutLong" is necessary here. Deleting resources is probably faster than creating new resources. Have you tried with shorter timeout?

Comment (sutaakar) [reply test/e2e/mnist_rayjob_raycluster_test.go:99]: hmm, I guess short timeout should be enough, will adjust the PR

Comment (sutaakar) [reply test/e2e/mnist_rayjob_raycluster_test.go]: Done.

**Review by jiripetrlik — APPROVED** [review]

**Pull request merged**